xwayland.c

```diff
@@ -167,9 +167,9 @@ handle_xwayland_surface_new(struct wl_listener *listener, void *data)
 	xwayland_view->xwayland_surface = xwayland_surface;
 
 	xwayland_view->map.notify = handle_xwayland_surface_map;
-	wl_signal_add(&xwayland_surface->events.map, &xwayland_view->map);
+	wl_signal_add(&xwayland_surface->surface->events.map, &xwayland_view->map);
```

Comment:
So… Unfortunately this won't work :/ the surface is not available before the `associate` event fires, and becomes invalid when the `dissociate` event fires…

Reply:
Ah sorry, had missed that.